lib/codebreaker/game.rb

```diff
@@ -0,0 +1,98 @@
+# frozen_string_literal: true
+
+module Codebreaker
+  class Game
+    include Validation
+    attr_accessor :secret_code, :attempts, :hints, :user_hints, :level_hints,
+                  :difficulty_name, :attempt_result, :attempts_total, :hints_total,
+                  :tmp_secret_code, :guess_won, :guess_loss
+
+    DIGIT_MIN_VALUE = 1
+    DIGIT_MAX_VALUE = 6
+    CODE_SIZE = 4
+    DIGIT_PLACE = '+'
+    DIGIT_PRESENCE = '-'
+
+    def initialize(difficulty)
+      @level_hints = []
+      @tmp_secret_code = []
+      @difficulty_name = difficulty[:level]
+      @attempts = difficulty[:attempts]
+      @attempts_total = difficulty[:attempts].clone
+      @hints = difficulty[:hints]
+      @hints_total = difficulty[:hints].clone
+      start_data
+    end
+
+    def start_data
+      @secret_code = [rand(1..6), rand(1..6), rand(1..6), rand(1..6)]
```

Comment:
use CODE_SIZE constant to handle secret_code length and use digi value range instead of 1..6